.dockerignore

```diff
@@ -0,0 +1,3 @@
+*
+!docker/
+!pkg/docker
```

Comment:
> That means that I needed to ignore both the upstream docker/ dir and the added pkg/docker dir.

This is the part that you got wrong :)

You are **un**ignoring those directories. This means, when you run `docker build`, the docker client will only pass `docker/` and `pkg/docker/` contents to the docker daemon to build the image. What I'm saying is you should only pass `pkg/docker/`, since we want to ignore the original docker stuff in the upstream repo.

To be more clear, you only need the stuff in `pkg/docker/` when building the docker image, hence you should only have this in the `.dockerignore` file:
```
*             # Ignore everything
!pkg/docker/  # but don't ignore the pkg/docker/ directory
```
(I'm not sure if comments are valid there, only including them for clarification)